### PR TITLE
part 3

### PR DESCRIPTION
**Pull request by GinaAntonini**

### CODE REVIEW

**Comment by GinaAntonini** [comment]
my delete button is working locally but it's not showing up on my deployed site. [gina's contacts](https://angular-contacts-495fa.firebaseapp.com/)

edit: just kidding it's working now

File: javascripts/AppConfig.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-app.run(function(FIREBASE_CONFIG){
+app.run(function($location, $rootScope, FIREBASE_CONFIG, AuthService){
   firebase.initializeApp(FIREBASE_CONFIG);
 });
 
 app.config(function($routeProvider){
   $routeProvider
     .when("/contacts/favorites", {
       templateUrl: 'partials/favorites.html',
-      controller: 'FavoritesCtrl'
```
Comment (rousell) [inline javascripts/AppConfig.js:11]: you don't need to be adding the commas here, since the last key value pair of an object shouldn't have a comma

**Review by rousell — COMMENTED** [review]
Hi @GinaAntonini, a couple of things. I am getting the following error in my console when clicking 'Log out': AuthService.logout is not a function. So you'll want to add that function to the AuthService. 

Additionally, I am currently able to access /contacts/new before logging in. You will want to add an event listener in your AppConfig app.run for `$rootScope.$on('$routeChangeStart'` to make sure that those paths are not exposed pre-authentication.

Let me know when you've made those changes and I'll review this again :) (and please let me know if you have any questions)

**Comment by GinaAntonini** [comment]
@rousell I think I am all set. Can you take a look?

**Comment by rousell** [comment]
@GinaAntonini Nice, this looks pretty good! the only thing that's missing is the validation on the form that the fields are completed: `
A user should not be able to submit a contact without putting information in each input on the form (check out ng-validate to help with this)` you can use this as a reference: https://scotch.io/tutorials/angularjs-form-validation

**Comment by GinaAntonini** [comment]
@rousell I put the validation stuff on part4

**Comment by rousell** [comment]
👍 noted! merging in

**Pull request merged**